lib/engine/game/g_1893/step/starting_package_forced_auction.rb

```diff
@@ -0,0 +1,192 @@
+          end
+
+          def help
+            return 'Select a company to auction' unless @auctioning
+
+            "Buy or Pass. If everyone pass price is lowered by #{@game.format_currency(10)} and a new buy/pass " \
+              'opportunity is presented. This continues until someone buys it or when price would go below 50% ' \
+              'of the value in which case the  player that selected the company is force to buy it. ' \
+              'Then next player in SR order gets to select a new company to be auctioned. And so on until all ' \
+              'companies are sold. Then the game continues with SR2. Note! Minors bought in this way will get the '\
+              "purchase price as treasury, or #{@game.format_currency(100)}, whichever is highest."
+          end
+
+          def actions(entity)
+            return [] if available.empty?
+            return [] unless entity == current_entity
+
+            return %w[bid] unless @auctioning
+
+            ACTIONS
+          end
+
+          def setup
+            @reduction_step = 10
+            @auction_triggerer = nil
+
+            setup_auction
+          end
+
+          def available
+            @auctioning ? [@auctioning] : @game.buyable_companies
+          end
+
+          def process_pass(action)
+            return unless @auctioning
+
+            target = @auctioning
+            entity = action.entity
+
+            pass_auction(entity)
+            @auctioning = target
+
+            if entities.all?(&:passed?)
+              if (target.discount + @reduction_step) > target.value / 2
+                force_purchase(@auction_triggerer, target)
+              else
+                all_passed!(target)
+              end
+              return
+            end
+
+            next_entity! if @auctioning
+          end
+
+          def pass_auction(entity)
+            entity.pass!
+
+            super
+          end
+
+          def next_entity!
+            @round.next_entity_index!
+            entity = entities[entity_index]
+            next_entity! if entity&.passed?
+          end
+
+          def process_bid(action)
+            return purchase(action) if @auctioning
+
+            selection_bid(action)
+            next_entity!
+          end
+
+          def min_bid(company)
+            return unless company
+
+            company.min_bid
+          end
+
+          def may_purchase?(_entity)
+            !!@auctioning
+          end
+
+          def may_choose?(_entity)
+            !@auctioning
+          end
+
+          def max_bid(entity, target)
+            may_purchase?(target) ? target.min_bid : entity.cash
+          end
+
+          def selection_bid(bid)
+            @auction_triggerer = bid.entity
+            target = bid_target(bid)
+
+            @game.log << "#{@auction_triggerer.name} selects #{target.name} for aution"
+            auction_entity(target)
+          end
+
+          def auctioning # rubocop:disable Style/TrivialAccessors
```

Comment:
Why implement this and disable the rubocop instead of adding `attr_reader :auctioning` at the top of the class?

Reply:
Copy paste

Fixed